### PR TITLE
WIP: ubuntu and node releases using cloudbuild.yaml and build.sh

### PR DESCRIPTION
**Pull request by aaron-prindle**
- [ ] get feedback on appropriate release flow
- [ ] add build triggers to launch the cloudbuild

### CODE REVIEW

File: reproducible/ubuntu/build.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (dlorenc) [inline reproducible/ubuntu/build.sh]: I'd drop the build script here. It doesn't really add anything, and we should be using triggers or running this ourselves.

**Comment by aaron-prindle** [comment]
Currently, the plan is to have 4 cloudbuild triggers based on tags.  The cloudbuild tags are:
ubuntu_16_0_4_release.*
ubuntu_16_0_4_build_release.*
node_8_5_0_ubuntu_release.*
node_8_5_0_ubuntu_build_release.*

so to release an image, someone uploads a tag:
`ubuntu_16_0_4_release_v1.0.0`
and the cloudbuild triggers creating the image:
gcr.io/gcp-runtimes/ubuntu_16_0_4:<COMMIT_SHA>

File: reproducible/ubuntu/BUILD [hunk 2]
```diff
@@ -20,27 +20,28 @@ pkg_tar(
     strip_prefix = "overlay/",
 )
 
-# 'ubuntu_vanilla' is docker image derived from a snapshot ubuntu rootfs
+# 'ubuntu_16_0_4_vanilla' is docker image derived from a snapshot ubuntu_16_0_4 rootfs
+>>>>>>> ubuntu and node images cloudbuild triggers using images_release.yaml
```
Comment (dlorenc) [inline reproducible/ubuntu/BUILD]: merge conflict?

File: reproducible/ubuntu/images_release.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+steps:
+  - name: 'gcr.io/cloud-builders/bazel'
+    args: ['run', '//reproducible/ubuntu:${_IMAGE}']
```
Comment (dlorenc) [inline reproducible/ubuntu/images_release.yaml]: I think these substitutions are the old envsubst style. Cloud build supports some now that are built in. What is _IMAGE supposed to be here?

File: reproducible/ubuntu/BUILD (hunk 2 above)
Comment (aaron-prindle) [reply reproducible/ubuntu/BUILD]: fixed

File: reproducible/ubuntu/images_release.yaml (hunk 3 above)
Comment (aaron-prindle) [reply reproducible/ubuntu/images_release.yaml]: IMAGE is the bazel-rule to run which shares a name with the image to be created.  I created this one cloudbuild rule and then I added the var in each of the cloud build triggers. So for the ubuntu_16_0_4_release_vX.Y.Z trigger -> _IMAGE=ubuntu_16_0_4
ubuntu_16_0_4_build_release_vX.Y.Z trigger -> _IMAGE=ubuntu_16_0_4_build
node_8_5_0_ubuntu_release_vX.Y.Z trigger -> _IMAGE=node_8_5_0_ubuntu
node_8_5_0_ubuntu_build_release_vX.Y.Z trigger -> _IMAGE=node_8_5_0_ubuntu_build

this way we can just have one cloudbuild.yaml.  We can also split it up.  I use one of the built in variables for the tagging, `${COMMIT_SHA}`

File: reproducible/ubuntu/test/BUILD [hunk 4]
```diff
@@ -11,7 +11,7 @@ structure_test(
 structure_test(
     name = "ubuntu_build_test",
     config = "//tests:ubuntu_build_test.yaml",
-    image = "//reproducible/ubuntu:ubuntu_build",
+    image = "//reproducible/ubuntu:ubuntu_16_0_4_build",
```
Comment (dlorenc) [inline reproducible/ubuntu/test/BUILD]: Looks like you missed the rename up on line 5:

```shell
ERROR: /home/travis/gopath/src/github.com/GoogleCloudPlatform/debian-docker/reproducible/ubuntu/test/BUILD:5:1: no such target '//reproducible/ubuntu:ubuntu': target 'ubuntu' not declared in package 'reproducible/ubuntu' defined by /home/travis/gopath/src/github.com/GoogleCloudPlatform/debian-docker/reproducible/ubuntu/BUILD and referenced by '//reproducible/ubuntu/test:ubuntu_test'
```

Comment (aaron-prindle) [reply reproducible/ubuntu/test/BUILD]: fixed

**Comment by aaron-prindle** [comment]
I have changed the cloudbuild triggers to run at every merge to master instead of using release tags

**Comment by aaron-prindle** [comment]
This should be ready now

**Pull request merged**